### PR TITLE
Depth Buffer Sharing Material Warning

### PR DESCRIPTION
**Pull request by Cameron-Micka**
## Overview
When enabling depth buffer sharing the stabilization of renderables greatly diminishes when a user looks at a renderable that has a material which does not write to the depth buffer. This has been a common pain point for HoloLens 2 users which don't understand why their holograms look distorted or unstable.

This pull request introduces a warning and "Fix Now" button on materials which are in a project that has depth buffer sharing enabled and do not write depth.

On the MRTK/Standard shader:
DepthWarning

On the MRTK/Text Mesh Pro shader:
DepthWarningTMP

If the "Fix Now" button will spawn a dialog confirming the user wants to change the material to write to the depth buffer.

Other changes include:

-  Moving the "Albedo Assigned At Runtime" option up the inspector for better visibility.
-  Bug fix, allowing the alpha cutoff slider to be adjusted in the  "Custom" render mode.

## Changes
- Fixes: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/4733

## Verification
To verify select a material that does not write depth (often transparent or additive materials) and select the "Fix Now" button and check if zWrite is now enabled.

### CODE REVIEW

File: Assets/MixedRealityToolkit/Inspectors/MixedRealityToolkitShaderGUIUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        {
+            public static readonly GUIContent DepthWriteWarning = new GUIContent("<color=yellow>Warning:</color> Depth buffer sharing is enabled for this project, but this material does not write depth. Enabling depth will improve reprojection, but may cause rendering artifacts in translucent materials.");
+            public static readonly GUIContent DepthWriteFixNowButton = new GUIContent("Fix Now", "Enables Depth Write For This Material");
+        }
+
+        /// <summary>
+        /// Checks if the project has depth buffer sharing enabled.
+        /// </summary>
+        /// <returns>True if the project has depth buffer sharing enabled, false otherwise.</returns>
+        public static bool IsDepthBufferSharingEnabled()
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit/Inspectors/MixedRealityToolkitShaderGUIUtilities.cs]: Did you copy this from the Optimize window helper? If so, can you not just call that utils?

Comment (Cameron-Micka) [reply Assets/MixedRealityToolkit/Inspectors/MixedRealityToolkitShaderGUIUtilities.cs]: I wish I could have, but there was just a settor (not a gettor) for depth buffer sharing. But, I did gleam how you accessed the project properties pre-2019 from there!

File: Assets/MixedRealityToolkit/StandardAssets/Shaders/MixedRealityTextMeshPro.shader [hunk 2]
```diff
@@ -57,6 +58,7 @@ Properties {
 	_StencilReadMask	("Stencil Read Mask", Float) = 255
 	
 	_ColorMask			("Color Mask", Float) = 15
+    _ZWrite             ("Depth Write", Float) = 0
```
[the remark this answers is not in the document]
Comment (Cameron-Micka) [reply Assets/MixedRealityToolkit/StandardAssets/Shaders/MixedRealityTextMeshPro.shader]: Ah yes, this file original came from TMP. Does MRTK have a preference for tabs or spaces?

[the remark this answers is not in the document]
Comment (keveleigh) [reply Assets/MixedRealityToolkit/StandardAssets/Shaders/MixedRealityTextMeshPro.shader]: We've been using spaces for code, but it does look like several of our shaders still have tabs. We should probably convert them?

**Pull request merged**